[cnsmap branch] Brief tidy-up of the Threat Modeling Section (#640)
cncf/tag-security: content/threat-modelling.md

category: "Security Assurance"
---

> For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. While there are many threat modeling techniques they share several core characteristics. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores, and [security boundaries](https://csrc.nist.gov/glossary/term/Security_boundary). Once boundaries have been established and the relevant elements of the system partitioned within those boundaries, the next step is to model how these elements interact with special attention paid to any interactions that cross security boundaries. Tools like [OWASP Threat Dragon](https://owasp.org/www-project-threat-dragon/) and [Microsoft Threat Modeling Tool](https://docs.microsoft.com/en-gb/azure/security/develop/threat-modeling-tool) help guide users through this process and can perform analysis of the resulting diagrams to automatically detect some threats.

Author: Thinking about it, if you look at the text in : https://csrc.nist.gov/glossary/term/Security_boundary I don't think it's even remotely helpful in understanding what a security boundary is in this context.

Reviewer: its a primary source definition, but i have to agree it lacks appropriate context here correcting with a better O'Reilly link (for sanity, we attempt to link to a source for defining content so we don't have to maintain the text)
```suggestion
For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. While there are many threat modeling techniques they share several core characteristics. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores, and [security boundaries](https://www.oreilly.com/library/view/cissp-certified-information/9780470276884/9780470276884_security_boundaries.html#:~:text=A%20security%20boundary%20is%20the,a%20LAN%20and%20the%20Internet.). Once boundaries have been established and the relevant elements of the system partitioned within those boundaries, the next step is to model how these elements interact with special attention paid to any interactions that cross security boundaries. Tools like [OWASP Threat Dragon](https://owasp.org/www-project-threat-dragon/) and [Microsoft Threat Modeling Tool](https://docs.microsoft.com/en-gb/azure/security/develop/threat-modeling-tool) help guide users through this process and can perform analysis of the resulting diagrams to automatically detect some threats.
```

> For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. There are many threat modeling techniques but they share commonalities. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores and security boundaries. A security boundary is any "dividing line of trust" - this can be physical (e.g a perimeter fence), it can be topological (e.g a network boundary) or it can be logical (e.g an authorization boundary). Once boundaries have been established and the relevant elements of the system partitioned within those boundaries, the next step is to model how these elements of the system interact with special attention being paid to any interactions that cross security boundaries. Tools like [OWASP Threat Dragon](https://owasp.org/www-project-threat-dragon/) and [Microsoft Threat Modeling Tool](https://docs.microsoft.com/en-gb/azure/security/develop/threat-modeling-tool) help guide users through this process and can perform analysis of the resulting diagrams to automatically detect some threats.

Reviewer: ```suggestion
For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. While there are many threat modeling techniques they share several core characteristics. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores, and [security boundaries](https://csrc.nist.gov/glossary/term/Security_boundary). Once boundaries have been established and the relevant elements of the system partitioned within those boundaries, the next step is to model how these elements interact with special attention paid to any interactions that cross security boundaries. Tools like [OWASP Threat Dragon](https://owasp.org/www-project-threat-dragon/) and [Microsoft Threat Modeling Tool](https://docs.microsoft.com/en-gb/azure/security/develop/threat-modeling-tool) help guide users through this process and can perform analysis of the resulting diagrams to automatically detect some threats.
```

- Tampering of an API server config file or certificate could result in failed API server restart or mutual TLS authentication failures
- Repudiation of actions of an attacker because of disabled or misconfigured API auditing could result in a lack of evidence of a potential attack
- Information disclosure is possible if an attacker compromises a running workload and is able to exfiltrate data to an external entity
- Denial of Service (DoS) resulting from a pod that does not have resource limits applied therefore consumes the entire node level CPU and memory, worker node is then lost
- Elevation of privilege could happen if a pod is running with unrestricted or higher privileged pod security policy or by modifying the security context of a pod or a container

Threat actors to consider for cloud native security are consistent with existing threat modeling:

- Malicious insider - An actor with malicious intent and with authorization to perform actions within the modelled system.
- Uninformed insider - An actor that with authorization to perform actions within the modelled system (assume anyone can be duped).

> - Malicious outsider - An actor outside of the system. They could be attacking via the internet, via supply chain, via physical perimeter etc.

Reviewer: uninformed outsider was removed.   any reason?  it is possible users outside of the system may inadvertently introduce issues.  For example, high school students writing scripts that create an unclosed loop which performs a DoS against their college or perhaps a banking website.

Author: I see your point, these actors exist. I'm not sure how they're useful in a threat model though. An damaging action available (like the DoS you describe above) to an Uninformed Outside is also available to someone with malicious intent. Likewise the mechanisms for constraining "accidental bad behavior" are likely to be the same as those for a Malicious actor.

Reviewer: it may beneficial to add a statement to the effect of: "for the purposes of threat modeling, attack avenues of an uninformed outsider are represented in a malicious outsider without intent" for both this and the CNSWP.

Author: Done, at least a variation of that in my own words (line 30)

category: "Security Assurance"
---

> For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. While there are many threat modeling techniques they share several core characteristics. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores, and [security boundaries](https://www.oreilly.com/library/view/cissp-certified-information/9780470276884/9780470276884_security_boundaries.html#:~:text=A%20security%20boundary%20is%20the,a%20LAN%20and%20the%20Internet.). Once boundaries have been established and the relevant elements of the system partitioned within those boundaries, the next step is to model how these elements interact with special attention paid to any interactions that cross security boundaries. 

Reviewer 2: ```suggestion
For organizations adopting cloud native, a primary mechanism for identifying risks, controls and mitigations is to perform threat modeling. While there are many threat modeling techniques they share several core characteristics. All start with building a scoped representation of a system's architecture. This begins with identifying all important processes, data stores, and [security boundaries](https://www.oreilly.com/library/view/cissp-certified-information/9780470276884/9780470276884_security_boundaries.html). Once boundaries have been established and the relevant elements of the system are partitioned within them, the next step is to model how these elements interact with special attention paid to any interactions that cross security boundaries. 
```

> The below guidance is an enhancement of the four step OWASP threat modeling recommended for cloud native capabilities.

Reviewer 2: ```suggestion
The below guidance is an enhancement of the four step [OWASP threat modeling](https://owasp.org/www-community/Threat_Modeling) recommended for cloud native capabilities.
```

- Spoofing a cluster admin by stealing the authentication credentials via a social engineering attack
- Tampering of an API server config file or certificate could result in failed API server restart or mutual TLS authentication failures
- Repudiation of actions of an attacker because of disabled or misconfigured API auditing could result in a lack of evidence of a potential attack
- Information disclosure is possible if an attacker compromises a running workload and is able to exfiltrate data to an external entity
- Denial of Service (DoS) resulting from a pod that does not have resource limits applied therefore consumes the entire node level CPU and memory, worker node is then lost
- Elevation of privilege could happen if a pod is running with unrestricted or higher privileged pod security policy or by modifying the security context of a pod or a container

Threat actors to consider for cloud native security are consistent with existing threat modeling practices:

> - Malicious insider - An actor with malicious intent and with authorization to perform actions within the modeled system.
> - Uninformed insider - An actor that with authorization to perform actions within the modeled system (assume anyone can be duped).
> - Malicious outsider - An actor outside of the system. They could be attacking via the internet, via supply chain, via physical perimeter etc.

Reviewer 2: ```suggestion
- Malicious insider - An actor with malicious intent and with authorization to perform actions within the modeled system.
- Uninformed insider - An actor with authorization to perform actions within the modeled system (assuming anyone can be duped).
- Malicious outsider - An actor with malicious intent and outside of the system, able to launch attacks via the internet, supply chain, physical perimeter etc. without explicit authorization to perform actions within the modeled system.
```
